lib/model/message.dart

```diff
@@ -56,7 +61,11 @@ class MessageStoreImpl with MessageStore {
   }
 
   void dispose() {
-    for (final view in _messageListViews) {
+    // When a MessageListView is disposed, it removes itself from the set
+    // `MessageStoreImpl._messageListViews`, which is the exact collection this
+    // iterates on. We need to make a copy of the collection to avoid concurrent
+    // modifications.
+    for (final view in _messageListViews.toList()) {
```

Comment:
nit: Well, it isn't the exact collection this iterates on :) because we're avoiding doing that, for the reason this mentions. How about:

```suggestion
    // When a MessageListView is disposed, it removes itself from the Set
    // `MessageStoreImpl._messageListViews`. Instead of iterating on that Set,
    // iterate on a copy, to avoid concurrent modifications.
```